syncopy/datatype/methods/selectdata.py

```diff
@@ -363,6 +365,23 @@ def selectdata(data,
     selectMethod.compute(data, out, parallel=kwargs.get("parallel"),
                          log_dict=log_dct)
 
+    # Handle selection of waveform for SpikeData objects
+    if type(data) == spy.SpikeData and data.waveform is not None:
+        if inplace:
+            spy.log("Inplace selection of SpikeData with waveform not supported for the waveform.", level="INFO")
+        else:
+            fauxTrials = [data._preview_trial(trlno) for trlno in data.selection.trial_ids]
+            spikes_by_trial = [f.idx[0] for f in fauxTrials]
+            spike_idx = np.concatenate([np.array(x).ravel() for x in spikes_by_trial])
```

Comment:
For wild sub-selections of channels/unit this will generate inhomogenuous index arrays, which as we know now are very very slow with `h5py`. However this should be also true for ordinary `SpikeData` selections, so I don't think we can do better here. Except for selecting whole trials, that's what we have those `_trialslice`'s now for, What's your take on this @kshapcott ?